test/parallel/test-trace-events-dns.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+const path = require('path');
+const tmpdir = require('../common/tmpdir');
+const fs = require('fs');
+const util = require('util');
+
+if (!common.isMainThread)
+  common.skip('process.chdir is not available in Workers');
+
+const traceFile = 'node_trace.1.log';
+
+const names = [
+  'lookup',
+  'lookupService',
+  'reverse',
+  'resolveAny',
+  'resolve4',
+  'resolve6',
+  'resolveCname',
+  'resolveMx',
+  'resolveNs',
+  'resolveTxt',
+  'resolveSrv',
+  'resolvePtr',
+  'resolveNaptr',
+  'resolveSoa'
+];
+
+tmpdir.refresh();
+process.chdir(tmpdir.path);
+
+const tests = 'const dns = require("dns");' +
+'const options = {' +
+'  family: 4,' +
+'  hints: dns.ADDRCONFIG | dns.V4MAPPED,' +
+'};' +
+'dns.lookup("example.com", options, (err, address, family) => {});' +
+'dns.lookupService("127.0.0.1", 22, (err, hostname, service) => {});' +
+'dns.reverse("8.8.8.8", (err, hostnames) => {});' +
+'dns.resolveAny("example.com", (err, res) => {});' +
+'dns.resolve4("example.com", (err, res) => {});' +
+'dns.resolve6("example.com", (err, res) => {});' +
+'dns.resolveCname("example.com", (err, res) => {});' +
+'dns.resolveMx("example.com", (err, res) => {});' +
+'dns.resolveNs("example.com", (err, res) => {});' +
+'dns.resolveTxt("example.com", (err, res) => {});' +
+'dns.resolveSrv("example.com", (err, res) => {});' +
+'dns.resolvePtr("example.com", (err, res) => {});' +
+'dns.resolveNaptr("example.com", (err, res) => {});' +
+'dns.resolveSoa("example.com", (err, res) => {});';
```

Comment:
does it make sense to wrap all these callbacks into `common.mustCall` semantic?

Reply:
Thanks for the comments. The callbacks are not significant in the trace log test case. There are existing test files that cover general dns features. The main focus here is to ensure an expected trace log is written to the trace output file. I did break the all-in-one test into individual tests so it will be easier to pinpoint the problem should an error occurs.